service/controller/resource/ipam/azure_machinepool_subnet_checker.go

```diff
@@ -58,44 +58,31 @@ func (c *AzureMachinePoolSubnetChecker) Check(ctx context.Context, namespace str
 		azureMachinePool = &expcapzv1alpha3.AzureMachinePool{}
 		err = c.ctrlClient.Get(ctx, objectKey, azureMachinePool)
 		if err != nil {
-			return false, microerror.Mask(err)
-		}
-
-		if !azureMachinePool.GetDeletionTimestamp().IsZero() {
-			c.logger.LogCtx(ctx, "level", "debug", "message", "AzureMachinePool is being deleted, skipping subnet allocation")
-			return false, nil
-		}
-	}
-
-	// Check if Cluster is being deleted. In that case we are skipping subnet allocation.
-	{
-		cluster, err := util.GetClusterFromMetadata(ctx, c.ctrlClient, azureMachinePool.ObjectMeta)
-		if err != nil {
-			return false, microerror.Mask(err)
-		}
-
-		if !cluster.GetDeletionTimestamp().IsZero() {
```

Comment:
In general these aren't necessary since OperatorKit takes care of the logic on deletion and in the future case this specific case is actually actively harmful because this code needs to work in `EnsureDeleted()` as well :slightly_smiling_face: